### DISCUSSION TITLE
How to get options time and sale

### DISCUSSION THREAD

**Discussion by NotMoni** [knicola/tdameritradejs#34] [Q&A] [answered]
Hey I have this code

```js
 streamer.on('authenticated', () => {
      streamer.subsTimesaleOptions('.SPY210730C440', ['symbol', 'tradeTime', 'lastPrice', 'lastSize', 'lastSequence']);
    });

streamer.on('timesale', (data) => {
      console.log(data);
    });
```

But nothing prints out. How am I supposed to get the options time and sale ?

**Comment by NotMoni** [comment, marked as answer]
Oops, the README.md said, "The data streamer implements most of what the documentation talks about except Actives, Level 1 Options, and Level 2 order book. ".

When will this feature be implemented? It would be really helpful.

**Reply by knicola** [reply]
I have a draft sitting around in a local branch but never got around to completing the unit tests. I'll see if I can finish this off over the weekend. I'll keep you posted.

**Reply by knicola** [reply]
@NotMoni Hey, just had some time to revisit this. Though Level 1 Options is incomplete, you should be able to use Timesale Options. If I'm not mistaken the Options symbol you are using is incorrect and it should be along the lines of `<ticker>_<date>[C for call | P for put]<price>` = `SPY_210730C440`. I'll try to test this tomorrow since it's only made available during market hours.

**Reply by NotMoni** [reply]
Ahh alright. Thanks for the feedback.

**Reply by NotMoni** [reply]
I will also test out tommorow different styles to see what works.

**Reply by knicola** [reply]
@NotMoni Looking at the [option chain endpoint](https://developer.tdameritrade.com/option-chains/apis/get/marketdata/chains) the symbol format I mentioned is correct, but timesale options service does not return any data. It's possible that TDA disabled the service. I was never able to get through to their API support before but I'll see if I can get a confirmation from them. Meanwhile, I'll get back to finishing up with the Level 1 Options subscription.

**Reply by NotMoni** [reply]
Oh alright. Thank you for the update.
